MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/MatrixApiActivity.java

```diff
@@ -113,23 +118,32 @@ public boolean onMarkerClick(@NonNull Marker marker) {
     });
   }
 
-  private int getClickedMarkerNumInPositionList(Marker clickedMarker) {
-    int clickedMarkerIndexPositionInList = -1;
-    if (clickedMarker != null) {
-      for (Marker singleMarker : mapboxMap.getMarkers()) {
-        if (singleMarker == clickedMarker) {
-          clickedMarkerIndexPositionInList = mapboxMap.getMarkers().indexOf(singleMarker);
+  @Override
+  public boolean onMapClick(@NonNull LatLng point) {
+    List<Feature> renderedStationFeatures = mapboxMap.queryRenderedFeatures(
+      mapboxMap.getProjection().toScreenLocation(point), LAYER_ID);
+    if (!renderedStationFeatures.isEmpty()) {
+      Point pointOfSelectedStation = (Point) renderedStationFeatures.get(0).geometry();
+      if (pointOfSelectedStation != null) {
+        String selectedBoltFeatureName = renderedStationFeatures.get(0).getStringProperty(STATION_NAME_PROPERTY);
+        List<Feature> featureList = featureCollection.features();
+        for (int i = 0; i < featureList.size(); i++) {
+          if (featureList.get(i).getStringProperty(STATION_NAME_PROPERTY).equals(selectedBoltFeatureName)) {
+            makeMapboxMatrixApiCall(i);
```

Comment:
The request could've been made during initialization and cached to be reused on map click.